packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -79,18 +80,24 @@ export async function generateSideloadFile(app: OfficeApp, manifest: ManifestInf
     if (!extEntry) {
       throw new ExpectedError("webextension was not found.");
     }
-
+    
     const webExtensionXml = templateZip
-      .readAsText(extEntry)
-      .replace(/00000000-0000-0000-0000-000000000000/g, manifest.id)
-      .replace(/1.0.0.0/g, manifest.version);
+    .readAsText(extEntry)
+    .replace(/00000000-0000-0000-0000-000000000000/g, manifest.id)
+    .replace(/1.0.0.0/g, manifest.version);
 
     templateZip.getEntries().forEach(function (entry) {
       var data: Buffer = entry.getData();
-      if (entry == extEntry) {
+      if (entry == extEntry && manifest.manifestType == ManifestType.XML) {
         data = Buffer.from(webExtensionXml);
       }
       outZip.addFile(entry.entryName, data, entry.comment, entry.attr);
+      
+      const webExtensionFolderPath = webExtensionPath.replace(/(\/[^\/]*)$/, '');
```

Comment:
Is this removing a trailing slash or backslash? The regex is hard to understand with the escaping. Perhaps a comment to explain what it is supposed to be doing would help.

Reply:
That or do something like webExtensionPath.substr(0, webExtensionPath.lastIndexOf("/"))

Reply:
Have replaced it with this method.